src/libmongoc/tests/test-mongoc-uri.c

```diff
@@ -255,6 +255,25 @@ test_mongoc_uri_new (void)
    ASSERT (!mongoc_uri_new (
       "mongodb://localhost/?" MONGOC_URI_HEARTBEATFREQUENCYMS "=499"));
 
+   /* timeoutMS */
+   uri = mongoc_uri_new ("mongodb://localhost/?" MONGOC_URI_TIMEOUTMS "=3000");
+   ASSERT (uri);
+   ASSERT_CMPINT64 (3000, ==, mongoc_uri_get_option_as_int64 (uri, MONGOC_URI_TIMEOUTMS, 0));
```

Comment:
Can you add a check for int32 as well? `mongoc_uri_set_option_as_int32` and `mongoc_uri_get_option_as_int32` should both work. There is some special behavior for `get_option_as_int32` to check for values within int32 range.